### PR TITLE
Talk about how to implement a DataManipulator on a block.

### PR DESCRIPTION
**Pull request by JBYoshi**
Based on https://github.com/SpongePowered/SpongeCommon/pull/1425.

### CODE REVIEW

File: source/contributing/implementation/datamanipulator.rst [hunk 1]
```diff
@@ -522,6 +524,75 @@ can be registered by subsequent calls of the ``registerValueProcessor()`` method
...
+    public Optional<BlockState> getStateWithData(IBlockState blockState, ImmutableDataManipulator<?, ?> manipulator) {
+        if (manipulator instanceof ImmutableDirectionalData) {
+            final Direction direction = ((ImmutableDirectionalData) manipulator).direction().get();
+            final EnumFacing facing = DirectionResolver.getFor(direction);
+            return Optional.of((BlockState) blockState.withProperty(BlockHorizontal.FACING, facing));
+        }
+        return super.getStateWithData(blockState, manipulator);
+    }
+
+``getStateWithValue()`` is the equivalent of ``getStateWithData()``, but works with single ``Key``s.
```
Comment (Inscrutable) [inline source/contributing/implementation/datamanipulator.rst]: You need to escape the backtick and add a space to pluralise, eg. ``DataManipulator``\ s
I think that's what breaks the formatting.

Comment (JBYoshi) [inline source/contributing/implementation/datamanipulator.rst]: That works. Thanks!

**Comment by Inscrutable** [comment]
This PR should conflict (very slightly) with the (0ne word) change I just made in the OCD fix omnibus PR. Something odd is afoot here.

**Comment by gabizou** [comment]
Code wise, this looks ok (from an implementation standpoint).

**Pull request merged**